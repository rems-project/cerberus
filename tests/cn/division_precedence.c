int divide_no_parenthesis ()
/*@ ensures return == 16i32; @*/
{
        return 8 + 32 / 4;  // 8 + (32 / 4)
}

int multiply_then_divide ()
/*@ ensures return == 28i32; @*/
{
        return 10 * 14 / 5; // (10 * 14) / 5
}

int divide_multiply_add_subtract ()
/*@ ensures return == 29i32; @*/
{
        return 20 / 10 * 30 - 100 / 2 + 4 * 5 - 10 * 10 / 100; // 2 * 30 - 50 + 20 - 1
}
