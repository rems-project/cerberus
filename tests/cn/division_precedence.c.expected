return code: 0
[1/3]: divide_no_parenthesis -- pass
[2/3]: multiply_then_divide -- pass
[3/3]: divide_multiply_add_subtract -- pass
